typed-builder-macro/src/enum_info.rs

```diff
@@ -0,0 +1,158 @@
+use convert_case::{Case, Casing};
+use proc_macro2::{Span, TokenStream};
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse::Error, parse_quote, punctuated::Punctuated, Token};
+
+use crate::builder_attr::{IntoSetting, TypeBuilderAttr};
+use crate::struct_info::StructInfo;
+
+pub struct EnumInfo<'a> {
```

Comment:
I don't like this struct. Unlike `StructInfo` and `FieldInfo`, which hold a parsed representation of the struct/field ready to create the builder code from, this `EnumInfo` here holds direct references to the AST. It's `new` method does no parsing - only verification.